### DISCUSSION TITLE
Copy-PnPFile fails without an error message

### DISCUSSION THREAD

**Discussion by RSCA421** [pnp/powershell#3906] [Q&A]
I am trying to copy a document library to another document library in the same SP online (live) site with the following commands.  I have left out the connection code and the setting of variables like $sourceTarget for brevity' sake. 

 ```
$sourceItems = Get-PnPListItem -List $sourceList -PageSize 5000 | Where-Object {$_.FieldValues.FileDirRef -eq $sourceTarget }
foreach($item in $sourceItems){
        try{
                Write-Host "Copying $($item.FieldValues.FileRef)"
                Copy-PnPFile -SourceUrl $item.FieldValues.FileRef -TargetUrl $destinationTarget -Force -OverwriteIfAlreadyExists -IgnoreVersionHistory
             }
         catch{
             Write-Host "ERROR copying file $($item.FieldValues.FileRef)"
            }
        }

```

The copy within the live site ( Source & Destination ) fails with no message.
A copy within dev ( Source & Destination ) work. 
A copy from live ( Source) to dev (Destination ) works.   

I have 2 problems I am trying to fix

- How do I get the error message out?
- How do I fix that error? 

I have SharePoint Admin role in portal.azure.com
Scripts are enabled for the site concern
Powershell is running PSVersion                      5.1.22621.2506     
PNP.Powershell is , I believe 1.7.0

**Comment by Barbarur** [comment]
I would recommend you to update the module to 1.12, that's the latest version that support PSv5. Just in case your issue was already fixed there.


Have you tried coping from dev ( Source) to live (Destination ) ? Just to confirm if any issue with live as destination.

**Comment by RSCA421** [comment]
Thank you for your reply.  I have tried copying from dev to live and that does not work.  
Live To Dev works
Dev To Dev works
My only concern about changing the version , is that it used to work Live To Live, & I'd be worried about introducing a version where other things might have got deprecated.  But I'll happily give it a try . Thanks again

**Reply by Barbarur** [reply]
Have to try to use live destination, but different location? It seem that's the issue. Maybe it's just a misspelling somewhere.

**Reply by RSCA421** [reply]
Thanks again. I don't think its a misspelling because the script had worked previously.  My references the destination sites are in the variable $destinationTarget .  The code has numerous versions of that , some pointing to the live site, and others to the test site, & I comment / uncomment these as necessary.
